external-crates/move/crates/move-compiler/src/shared/ide.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    diagnostics::Diagnostics, expansion::ast as E, ice, naming::ast as N, parser::ast as P,
+    shared::Name, typing::ast as T,
+};
+
+use move_ir_types::location::Loc;
+
+use std::collections::BTreeMap;
+
+#[derive(Debug, Clone)]
+pub struct IDEInfo {
+    pub exp_info: BTreeMap<Loc, ExpEntry>,
```

Comment:
I'm a bit worried about the longevity of the design here, as it forces a good handful of assumptions onto the structure of `ExpEntry`.

I would probably prefer something like `info: Vec<(Loc, ExpInfo)>`, and let the IDE sort it out. 
The mental model I have here is that the compiler can blindly tag locations with information, then the IDE can come along and sort them all out. 
This isn't too different to what we do with Diags, we have a big vector we push to, that we then muck around with when it comes time to report